python/cudf/CMakeLists.txt

```diff
@@ -74,42 +65,32 @@ include(rapids-cython)
 if(NOT cudf_FOUND)
   set(BUILD_TESTS OFF)
   set(BUILD_BENCHMARKS OFF)
+  set(CUDF_BUILD_TESTUTIL OFF)
+  set(CUDF_BUILD_STREAMS_TEST_UTIL OFF)
+  set(CUDA_STATIC_RUNTIME ON)
+
+  add_subdirectory(../../cpp cudf-cpp EXCLUDE_FROM_ALL)
 
-  set(_exclude_from_all "")
-  if(CUDF_BUILD_WHEELS)
-    # We don't build C++ tests when building wheels, so we can also omit the test util and shrink
-    # the wheel by avoiding embedding GTest.
-    set(CUDF_BUILD_TESTUTIL OFF)
-    set(CUDF_BUILD_STREAMS_TEST_UTIL OFF)
-
-    # Statically link cudart if building wheels
-    set(CUDA_STATIC_RUNTIME ON)
-
-    # Need to set this so all the nvcomp targets are global, not only nvcomp::nvcomp
-    # https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_TARGETS_GLOBAL.html#variable:CMAKE_FIND_PACKAGE_TARGETS_GLOBAL
-    set(CMAKE_FIND_PACKAGE_TARGETS_GLOBAL ON)
-
-    # Don't install the cuDF C++ targets into wheels
-    set(_exclude_from_all EXCLUDE_FROM_ALL)
-  endif()
-
-  add_subdirectory(../../cpp cudf-cpp ${_exclude_from_all})
-
-  if(CUDF_BUILD_WHEELS)
-    include(cmake/Modules/WheelHelpers.cmake)
-    get_target_property(_nvcomp_link_libs nvcomp::nvcomp INTERFACE_LINK_LIBRARIES)
-    # Ensure all the shared objects we need at runtime are in the wheel
-    add_target_libs_to_wheel(LIB_DIR cudf TARGETS arrow_shared nvcomp::nvcomp ${_nvcomp_link_libs})
-  endif()
-  # Since there are multiple subpackages of cudf._lib that require access to libcudf, we place the
-  # library in the cudf directory as a single source of truth and modify the other rpaths
-  # appropriately.
+  # libcudf targets are excluded by default above via EXCLUDE_FROM_ALL to remove extraneous
+  # components like headers from libcudacxx, but we do need the libraries. However, we want to
+  # control where they are installed to. Since there are multiple subpackages of cudf._lib that
+  # require access to libcudf, we place the library and all its dependent artifacts in the cudf
+  # directory as a single source of truth and modify the other rpaths appropriately.
   set(cython_lib_dir cudf)
-  install(TARGETS cudf DESTINATION ${cython_lib_dir})
+  include(cmake/Modules/WheelHelpers.cmake)
+  # TODO: This install is currently overzealous. We should only install the libraries that are
+  # downloaded by CPM during the build, not libraries that were found on the system.  However, in
+  # practice right this would only be a problem is if libcudf was not found but some of the
+  # dependencies were, and we have no real use cases where that happens.
```

Comment:
I don't want to try and fix this right now. I would like to manage the scikit-build-core transition first, then revisit the best way for us to handle install rules. Some anecdotal experiments suggest that the configure/build paths might be a bit different in scikit-build-core, so I don't want to do redundant work.